build_projects/dotnet-cli-build/build.sh

```diff
@@ -101,6 +101,15 @@ then
     ulimit -n 1024
 fi
 
+# Temporary logic to turn core dumps on so we can catch exit code 138 errors on OSX
+if [ "$(uname -s)" == "Darwin" ]; then
+  # If there are no core files already in /cores/, we'll enable core dump
+  # generation for this shell.
+  if [ "$(ls -A /cores)" ]; then 
```

Comment:
@eerhardt @gkhanna79 

The reason why the BuildTools change doesn't reset ulimit is that each of those scripts is invoked in its own shell and the ulimit change is limited to the shell it's made in (and any children). I haven't looked into how this CLI script is invoked or by whom, but that may or may not apply here.